### PR TITLE
fixes logical conversions

### PR DESCRIPTION
**Pull request by schweitzpgi**
add back a change lost in merge

This change fixes some more bugs in BLAS compilation.

### CODE REVIEW

File: flang/lib/Lower/ConvertExpr.cpp [hunk 1]
```diff
@@ -456,25 +452,32 @@ class ExprLowering {
...
     auto one = genLogicalConstantAsI1(context, true);
-    return builder.create<mlir::XOrOp>(getLoc(), logical, one).getResult();
+    auto val =
+        builder.create<fir::ConvertOp>(getLoc(), builder.getI1Type(), logical);
+    return builder.create<mlir::XOrOp>(getLoc(), val, one);
   }
 
   template <int KIND>
   mlir::Value genval(const Fortran::evaluate::LogicalOperation<KIND> &op) {
     // Request operands to be generated as `i1` and restore after this scope.
```
Comment (jeanPerier) [inline flang/lib/Lower/ConvertExpr.cpp:462]: I think this comment is outdated.

**Comment by jeanPerier** [comment]
Thanks, looks like a safer approach to me.

File: flang/lib/Lower/ConvertExpr.cpp (hunk 1 above)
Comment (schweitzpgi) [reply flang/lib/Lower/ConvertExpr.cpp:462]: Oops. +1

**Comment by schweitzpgi** [comment]
Ripping out the stop gap might expose some bugs that I haven't caught yet. But this looks like it will work for BLAS and LAPACK so far.

**Pull request merged**